python/bucketbase/cached_immutable_bucket.py

```diff
@@ -7,28 +7,42 @@
 from bucketbase.errors import DeleteError
 from bucketbase.fs_bucket import AppendOnlyFSBucket
 from bucketbase.ibucket import ShallowListing, IBucket, ObjectStream
+from bucketbase import namedlock
 
 
 class CachedImmutableBucket(IBucket):
-    def __init__(self, cache: IBucket, main: IBucket) -> None:
+    def __init__(self, cache: IBucket, main: IBucket, lock_manager: namedlock.NamedLockManager = None) -> None:
         self._cache = cache
         self._main = main
+        self._lock_manager = lock_manager or namedlock.ThreadLockManager()
```

Comment:
I'd say that here's a bug, as using the threading synchronization mecanism would make this class fail when used with a cache used by multiple processes, like our main scenario - using the FileSystem as cache